service/storage_proxy.cc

```diff
@@ -1788,11 +1788,11 @@ storage_proxy::storage_proxy(distributed<database>& db, storage_proxy::config cf
                        sm::description("number of currently throttled write requests")),
     });
 
-    if (cfg.hinted_handoff_enabled) {
+    if (!cfg.hinted_handoff_enabled.is_disabled_for_all()) {
         const db::config& dbcfg = _db.local().get_config();
         supervisor::notify("creating hints manager");
-        slogger.trace("hinted DCs: {}", *cfg.hinted_handoff_enabled);
-        _hints_manager.emplace(dbcfg.hints_directory(), *cfg.hinted_handoff_enabled, dbcfg.max_hint_window_in_ms(), _hints_resource_manager, _db);
+        slogger.trace("hinted DCs: {}", cfg.hinted_handoff_enabled);
```

Comment:
Please, adjust this log message.
The resulting message is going to be SUPER confusing, e.g. for all hints enabled it's going to be: "hinted DCs: enabled_kind=enabled, dcs={}".

Reply:
Note that your current series code is going to disable hints for view hints.

Reply:
Log message adjusted.

As for view hints, could you explain why this change would disable them? `_hints_for_views_manager` is initialized with a `host_filter` that accepts all DCs, so it should be enabled.

Reply:
I thought it was enabled with a default constructed hints_filter which originally was a "disable_all". If I misread it then I apologize.